### PR TITLE
Bug 1760393: Add metrics for collecting socket and hyperthreading enablement.

### PR DESCRIPTION
**Pull request by chambridge**
* cluster:hyperthread_enabled_nodes (kube_node_labels + label_node_hyperthread_enabled=true)
* cluster:capacity_cpu_cores_hyperthread_enabled:sum (similar to cluster:capacity_cpu_cores:sum but add categorizing by label_node_hyperthread_enabled)
* cluster:capacity_cpu_sockets_hyperthread_enabled:sum (similar to cluster:capacity_cpu_cores_hyperthread_enabled:sum but sockets)
* node_role_os_version_machine:cpu_capacity_cores:sum (updated to include label_node_hyperthread_enabled)
* node_role_os_version_machine:cpu_capacity_sockets:sum (similar to node_role_os_version_machine:cpu_capacity_cores:sum but sockets)

These metrics are intended to provide the necessary data for correct subscription management.

[cluster-hyperthread_enabled_nodes.txt](https://github.com/openshift/cluster-monitoring-operator/files/3768894/cluster-hyperthread_enabled_nodes.txt)
[cluster:capacity_cpu_cores_hyperthread_enabled:sum.txt](https://github.com/openshift/cluster-monitoring-operator/files/3768895/cluster.capacity_cpu_cores_hyperthread_enabled.sum.txt)
[cluster:capacity_cpu_sockets_hyperthread_enabled:sum.txt](https://github.com/openshift/cluster-monitoring-operator/files/3768896/cluster.capacity_cpu_sockets_hyperthread_enabled.sum.txt)
[node_role_os_version_machine-cpu_capacity_cores-sum.txt](https://github.com/openshift/cluster-monitoring-operator/files/3768897/node_role_os_version_machine-cpu_capacity_cores-sum.txt)
[node_role_os_version_machine-cpu_capacity_sockets-sum.txt](https://github.com/openshift/cluster-monitoring-operator/files/3768898/node_role_os_version_machine-cpu_capacity_sockets-sum.txt)

### CODE REVIEW

**Comment by LiliC** [comment]
/ok-to-test

uff 😅 those queries are indeed something

cc @openshift/openshift-team-monitoring

File: assets/prometheus-k8s/rules.yaml [hunk 1]
```diff
@@ -300,13 +315,60 @@ spec:
...
+        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled="true"})
+        * on(node) group_left sum(label_replace(node_cpu_info, "node", "$1", "instance",
+        "(.*)")) by (package, node)) or on(node) (cluster:hyperthread_enabled_nodes
+        * on(node) group_left sum(label_replace(node_cpu_info, "node", "$1", "instance",
+        "(.*)")) by (package, node))  or on(node) ((kube_node_labels unless on(node)
+        cluster:hyperthread_enabled_nodes) * on(node) group_left sum(label_replace(node_cpu_info,
+        "node", "$1", "instance", "(.*)")) by (package, node))) BY (label_node_openshift_io_os_id,
+        label_kubernetes_io_arch, label_node_hyperthread_enabled, label_node_role_kubernetes_io_master_infra,
+        label_node_role_kubernetes_io_master, label_node_role_kubernetes_io_infra)
+      record: node_role_os_version_machine:cpu_capacity_sockets:sum
```
Comment (LiliC) [inline assets/prometheus-k8s/rules.yaml]: I get no value when trying this on an existing cluster. It was a 4.1 cluster, is this specific to an openshift version?

Comment (chambridge) [reply assets/prometheus-k8s/rules.yaml]: @LiliC it is dependent on the work @pgier did for 4.3 to pull in `node_exporter` features that capture sockets and hyperthreading enablement:
https://bugzilla.redhat.com/show_bug.cgi?id=1760393#c1
https://bugzilla.redhat.com/show_bug.cgi?id=1760393#c2

These will need to be back ported to support earlier releases.

Comment (brancz) [reply assets/prometheus-k8s/rules.yaml]: What I see in the links in the description looks good to me.

**Pull request merged**